client/azure_client_set.go

```diff
@@ -63,117 +65,185 @@ func NewAzureClientSet(config AzureClientSetConfig) (*AzureClientSet, error) {
 
 	c := &clientConfig{
 		subscriptionID:          config.SubscriptionID,
+		partnerIdUserAgent:      fmt.Sprintf("pid-%s", config.PartnerID),
```

Comment:
Can we have this in a `key.go`file? Or does that make sense?

Reply:
I'm not sure what you mean, the partner id is not part of the `AzureConfig` CR. To me, it made sense to treat this value the same way we treat the subscription id. WDYT?

Reply:
Not used outside the file so I'd say we shouldn't move it to key.go.